pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -208,10 +208,68 @@ func (n *nodeHealthMap) set(name string, data *nodeHealthData) {
 	n.nodeHealths[name] = data
 }
 
+type evictionStatus int
+
+const (
+	unmarked = iota
+	toBeEvicted
+	evicted
+)
+
+type nodeEviction struct {
+	status evictionStatus
+	pods   map[types.UID]*v1.Pod
+}
+
+// nodeEvictionMap is NOT thread-safe.
+type nodeEvictionMap map[string]*nodeEviction
```

Comment:
nodeEvictionMap may grow indefinitely - we need a mechanism to clean it.

Not sure what's the best way - removing on Node removal may not be enough, because some pods may potentially be processed later?
Something you need to figure out.

Reply:
- deletetionStamp added to eviction map
- when node is deleted, deletionStamp is set
- if deletetionStamp is 5m old, node entry is removed.